app/pages/project/project-page.spec.js

```diff
@@ -130,4 +132,36 @@ describe('ProjectPage', () => {
       expect(fieldguide).to.have.lengthOf(1);
     });
   });
+
+  describe('on component lifecycle', () => {
+    const sugarClientSubscribeSpy = sinon.spy(sugarClient, 'subscribeTo');
+    const sugarClientUnsubscribeSpy = sinon.spy(sugarClient, 'unsubscribeFrom');
+    const channel = `project-${project.id}`;
+    let wrapper;
+    beforeEach(() => {
+      wrapper = mount(
+        <ProjectPage project={project}>
+          <Page />
+        </ProjectPage>
+      );
+    });
+
+    it('subscribes the user to the sugar project channel on mount', () => {
+      expect(sugarClientSubscribeSpy.calledWith(channel)).to.equal(true);
+    });
+
+    it('unsubscribes the user from the sugar project channel on unmount', () => {
+      wrapper.unmount();
+      expect(sugarClientUnsubscribeSpy.calledWith(channel)).to.equal(true);
+    });
+
+    it('unsubscribes old project and subscribes new project on project change', () => {
+      const newProject = {id: '999', title: 'fake project', slug: 'owner/name'};
+      const newChannel = `project-${newProject.id}`;
+      sugarClientSubscribeSpy.reset()
+      wrapper.setProps({ project: newProject });
+      expect(sugarClientSubscribeSpy.calledWith(newChannel)).to.equal(true);
+      expect(sugarClientUnsubscribeSpy.calledWith(channel)).to.equal(true);
```

Comment:
I think `calledWith` matches against any call in the spy's `calls` array, so this will pass from the unsubscribe in the previous test. Reset should fix it.